web_monitoring/html_diff_render.py

```diff
@@ -187,8 +187,74 @@
 MAX_SPACERS = 2500
 
 
+class WaybackUrlComparator:
+    matcher = re.compile(r'web/\d{14}(im_)?/(https?://)?(www.)?')
+    url_schema_matcher = re.compile(r'(https?\:\/\/)?(www\.)?')
+
+    def strip_url_schema(self, url):
+        schema_match = self.url_schema_matcher.search(url)
+        if schema_match:
+            return url[schema_match.end():]
+
+    def compare(self, url_a, url_b):
+        match_a = self.matcher.search(url_a)
+        match_b = self.matcher.search(url_b)
+        if match_a and match_b:
+            url_a = self.strip_url_schema(url_a[match_a.end():])
+            url_b = self.strip_url_schema(url_b[match_b.end():])
+            return url_a == url_b
+        else:
+            if hasattr(url_a, 'strict_urls'):
+                url_a.strict_urls = None
+            if hasattr(url_b, 'strict_urls'):
+                url_b.strict_urls = None
+            return url_a == url_b
+
+
+class WaybackUkUrlComparator(WaybackUrlComparator):
+    matcher = re.compile(r'https://www\.webarchive\.org\.uk/wayback/en/archive/\d{14}mp_/')
```

Comment:
> Is mp_ a valid suffix that’s not documented (or documented somewhere I don’t know about)? Is it particular to the UK?

I didn't find any documentation about it but when looking at the HTML code UKWA returns, if a link has the `_mp` attribute, the "raw" snapshot as it was saved is returned. And all their links have it. Otherwise, the source link is rendered in an iframe using JS, and is thus undiffable.

It it specific to the UKWA.